src/3rd_party.c

```diff
@@ -0,0 +1,173 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2019 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#define _GNU_SOURCE
+#include "3rd_party.h"
+#include "config.h"
+#include "lib/log.h"
+#include "swupd.h"
+
+static char *cmdline_option_repo = NULL;
+
+#define CHOOSE_ADD 9000
+#define CHOOSE_LIST 9001
+#define CHOOSE_REMOVE 9003
+static bool cmdline_option_add;
+static bool cmdline_option_list;
+static bool cmdline_option_remove;
+
+#ifdef THIRDPARTY
+
+void get_repo_path(char **repo_path, char *repo_name)
+{
+	string_or_die(repo_path, "%s%s", THIRDPARTY_REPO_PREFIX, repo_name);
```

Comment:
Use sys_path_join here to make sure we are going to have a '/' separator

Reply:
Makes sense will change in next push